UnitsNet/UnitConverter.cs

```diff
@@ -419,23 +419,51 @@ public static double ConvertByAbbreviation(double fromValue, string quantityName
         {
-            if (!TryGetUnitType(quantityName, out Type? unitType))
-                throw new UnitNotFoundException($"The unit type for the given quantity was not found: {quantityName}");
-
-            var cultureInfo = CultureHelper.GetCultureOrInvariant(culture);
-
-            var fromUnit = UnitsNetSetup.Default.UnitParser.Parse(fromUnitAbbrev, unitType, cultureInfo); // ex: ("m", LengthUnit) => LengthUnit.Meter
-            var fromQuantity = Quantity.From(fromValue, fromUnit);
+            return ConvertByAbbreviation(fromValue, quantityName, fromUnitAbbrev, toUnitAbbrev, CultureHelper.GetCultureOrInvariant(culture));
+        }
 
-            var toUnit = UnitsNetSetup.Default.UnitParser.Parse(toUnitAbbrev, unitType, cultureInfo); // ex:("cm", LengthUnit) => LengthUnit.Centimeter
-            return fromQuantity.As(toUnit);
+        /// <summary>
+        ///     Convert between any two quantity units by their abbreviations, such as converting a "Length" of N "m" to "cm".
+        ///     This is particularly useful for creating things like a generated unit conversion UI,
+        ///     where you list some selectors:
+        ///     a) Quantity: Length, Mass, Force etc.
+        ///     b) From unit: Meter, Centimeter etc if Length is selected
+        ///     c) To unit: Meter, Centimeter etc if Length is selected
+        /// </summary>
+        /// <param name="fromValue">
+        ///     Input value, which together with <paramref name="fromUnitAbbrev" /> represents the quantity to
+        ///     convert from.
+        /// </param>
+        /// <param name="quantityName">The invariant quantity name, such as "Length". Does not support localization.</param>
+        /// <param name="fromUnitAbbrev">The abbreviation of the unit in the given culture, such as "m".</param>
+        /// <param name="toUnitAbbrev">The abbreviation of the unit in the given culture, such as "m".</param>
+        /// <param name="formatProvider">
+        ///     The format provider to use for lookup. Defaults to <see cref="System.Globalization.CultureInfo.CurrentCulture" />
+        ///     if null.
+        /// </param>
+        /// <example>double centimeters = ConvertByName(5, "Length", "m", "cm"); // 500</example>
+        /// <returns>Output value as the result of converting to <paramref name="toUnitAbbrev" />.</returns>
+        /// <exception cref="QuantityNotFoundException">
+        ///     Thrown when no quantity information is found for the specified quantity name.
+        /// </exception>
+        /// <exception cref="UnitNotFoundException">No units match the abbreviation.</exception>
+        /// <exception cref="AmbiguousUnitParseException">More than one unit matches the abbreviation.</exception>
+        public static double ConvertByAbbreviation(double fromValue, string quantityName, string fromUnitAbbrev, string toUnitAbbrev, IFormatProvider? formatProvider)
```

Comment:
I've added an `IFormatProvider?` overload and deprecated the whole `CultureHelper` class:

> string -> CultureInfo conversions are not in the scope of UnitsNet

If you want to- we could also make this for `v5` and have the `string?` overload directly removed in `v6`.

Reply:
Oh I think I see what the deal is- the `UnitParser` doesn't really care about the `IFormat` part of the `IFormatProvider`, just the `CultureName`.. I wonder if there is a more _standard_ way of dealing with things that depend solely on the language / region (and not that number-formatting stuff).

Reply:
Given that we ultimately depend on the `ResourceManager`, and it takes a `CultureInfo?` as it's second parameter, I think the right way (another breaking change) would be to make everything that doesn't involve formatting the number (i.e. everything in the `UnitParser`, `UnitAbbreviationsCache`, as well as this new overload that I just added) use a `CultureInfo?` (with the rest of the code safe-casting the `IFormatProvider?`)..